src/_guides/language/effective-dart/design.md

```diff
@@ -1005,79 +1091,260 @@ var names = people.map((Person person) {
+    }
+  }
+}
+{% endprettify %}
+
+
+### DON'T specify a return type for a setter.
+
+Setters always return `void` in Dart. Writing the word is pointless.
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (avoid_return_types_on_setters)"?>
+{% prettify dart %}
+void set foo(Foo value) { ... }
+{% endprettify %}
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (avoid_return_types_on_setters)"?>
+{% prettify dart %}
+set foo(Foo value) { ... }
+{% endprettify %}
+
+
+### DON'T use the legacy typedef syntax.
+
+Dart has two notations for defining a named typedef for a function type. The
+original syntax looks like:
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (old-typedef)"?>
+{% prettify dart %}
+typedef int Comparison<T>(T a, T b);
+{% endprettify %}
+
+It has a couple of problems:
+
+*   There is no way to assign a name to a *generic* function type. In the above
+    example, the typedef itself is generic. If you reference `Comparison` in
+    your code, without a type argument, you implicitly get the function type
+    `int Function(dynamic, dynamic)`, *not* `int Function<T>(T, T)`. This
+    doesn't come up in practice often, but it matters in certain corner cases.
+
+*   A single identifier in a parameter is interpreted as the parameter's *name*,
+    not its *type*. Given:
+
+    {:.bad-style}
+    <?code-excerpt "misc/lib/effective_dart/design_bad.dart (typedef-param)"?>
+    {% prettify dart %}
+    typedef bool TestNumber(num);
+    {% endprettify %}
+
+    Most users expect this to be a function type that takes a `num` and returns
+    `bool`. It is actually a function type that takes *any* object (`dynamic`)
+    and returns `bool`. The parameter's *name* (which isn't used for anything
+    except documentation in the typedef) is "num". This has been a
+    long-standing source of errors in Dart.
+
+The new syntax looks like:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (new-typedef)"?>
+{% prettify dart %}
+typedef Comparison = int Function<T>(T, T);
+{% endprettify %}
+
+The new syntax can express anything the old syntax could express and more, and
+lacks the error-prone misfeature where a single identifier is treated as the
+parameter's name. The same function type syntax after the `=` in the typedef is
+also allowed anywhere a type annotation may appear, giving us a single
+consistent way to write function types anywhere in a program.
+
+The old typedef syntax is still supported to avoid breaking existing code, but
+is deprecated.
+
+
+### PREFER inline function types over typedefs.
+
+In Dart 1, if you wanted to use a function type for a field, variable, or
+generic type argument, you had to first define a typedef for it. Dart 2 supports
+a function type syntax that can be used anywhere a type annotation is allowed:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (function-type)"?>
```

Comment:
There is a lot of code in the excerpt; consider adding highlighting of `Function` lexemes to help focus the reader's attention:

    <?code-excerpt "misc/..." replace="/Function/[!$&!]/g"?>

Reply:
Ah, good idea! What that, I think you're right that the comments are needless.